### PR TITLE
New-DbaDbTable - ensure schema is created if it doesn't exist

### LINKED ISSUE #7153: New-DbaDbTable - create the schema if it doesn't exist
### Summary of new feature

The New-DbaDbTable command currently will return an error if the caller specifies a schema that does not exist. This feature request would allow for the schema to be created automatically.

### Proposed technical details (if applicable)
Modify the New-DbaDbTable command so that it checks if the schema exists and if not then create it.

### Latest version of dbatools as of writing
1.0.139

### PR DESCRIPTION
**Pull request by lancasteradam**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #7153 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [x] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Ensure that the schema is created instead of returning an error to the caller.

### Approach
Minor change to create schema before creating the table.

### Commands to test
See the new integration tests.

### CODE REVIEW

**Comment by lancasteradam** [comment]
@jpomfret, just a quick note that I'll need to update this PR after #7178 is merged.

File: functions/New-DbaDbTable.ps1 [hunk 1]
```diff
@@ -455,16 +455,28 @@ function New-DbaDbTable {
                         $object.Columns.Add($sqlColumn)
                     }
 
+                    # user has specified a schema that does not exist yet
+                    if (-not ($db | Get-DbaDbSchema -Schema $schema -IncludeSystemSchemas)) {
+                        $schemaObject = New-Object -TypeName Microsoft.SqlServer.Management.Smo.Schema $db, $schema
```
Comment (jpomfret) [inline functions/New-DbaDbTable.ps1:460]: Can we use the New-DbaDbSchema function you created here?

Comment (lancasteradam) [reply functions/New-DbaDbTable.ps1:460]: Glad you asked, I was originally hoping to do that, but then noticed the -Passthru logic implies that a caller could be scripting out the definitions and not actually creating the objects. So, I decided to use the SMO Schema object which could then be used in either the -Passthru scenario or the default scenario when the objects are actually created. 

I'm open to changing it so let me know your thoughts on this.

Comment (potatoqualitee) [reply functions/New-DbaDbTable.ps1:460]: This looks great, thank you. If you decide with using the other command, you can submit a PR for that.

**Pull request merged**